hoomd/hpmc/shape_move.py

```diff
@@ -0,0 +1,273 @@
+# Copyright (c) 2009-2022 The Regents of the University of Michigan.
+# Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+"""Shape moves for a for alchemical simulations in extended ensembles."""
+
+import hoomd
+from hoomd.operation import _HOOMDBaseObject
+from . import _hpmc
+from hoomd.hpmc import integrate
+from hoomd.data.parameterdicts import ParameterDict, TypeParameterDict
+from hoomd.data.typeparam import TypeParameter
+import numpy
+
+
+class ShapeMove(_HOOMDBaseObject):
+    """Base class for all shape moves.
+
+    Args:
+        move_probability (`float`): Probability of performing a shape move.
+        See the  documentation of each derived class for a
+            descrition of how this parameter is interpreted in the context of
+            each shape move subclass.
+
+    Note:
+        See the documentation of each derived class for a list of supported
+        shapes.
+
+    Note:
+        This class should not be instantiated by users. The class can be used
+        for `isinstance` or `issubclass` checks.
+    """
+
+    _suported_shapes = None
+
+    def __init__(self, move_probability):
+        # Set base parameter dict for all shape_moves
+        param_dict = ParameterDict(move_probability=float(move_probability))
+        self._param_dict.update(param_dict)
+
+    def _attach(self):
+        integrator = self._simulation.operations.integrator
+        if not isinstance(integrator, integrate.HPMCIntegrator):
+            raise RuntimeError("The integrator must be a HPMC integrator.")
+        if not integrator._attached:
+            raise RuntimeError("Integrator is not attached yet.")
+
+        integrator_name = integrator.__class__.__name__
+        if integrator_name in self._suported_shapes:
+            self._move_cls = getattr(_hpmc,
+                                     self.__class__.__name__ + integrator_name)
+        else:
+            raise RuntimeError("Integrator not supported")
+        self._cpp_obj = self._move_cls(
+            self._simulation.state._cpp_sys_def,
+            self._simulation.operations.integrator._cpp_obj)
+        super()._attach()
+
+
+class Elastic(ShapeMove):
+    """Apply scale and shear shape moves to particles with an energy penalty.
+
+    Args:
+        stiffness (:py:mod:`hoomd.variant.Variant`): Shape stiffness against
+            deformations.
+
+        mc (`type` or `hoomd.hpmc.integrate.HPMCIntegrator`): The class of
+            the MC shape integrator or an instance (see `hoomd.hpmc.integrate`)
+            to use with this elastic shape. Must be a compatible class.
+
+        move_probability (`float`, optional): Fraction of scale to shear
+            moves (**default**: 0.5).
+
+    .. rubric:: Shape support.
+
+    The following shapes are supported:
+        * `hoomd.hpmc.integrate.ConvexPolyhedron`
+        * `hoomd.hpmc.integrate.Ellipsoid`
```

Comment:
```suggestion
    The following shapes are supported:
    
    * `hoomd.hpmc.integrate.ConvexPolyhedron`
    * `hoomd.hpmc.integrate.Ellipsoid`
```
Format this as a list, not a definition list.